src/transformers/pipelines/token_classification.py

```diff
@@ -251,23 +257,62 @@ def __call__(
               exists if the offsets are available within the tokenizer
         """
 
-        _inputs, offset_mapping = self._args_parser(inputs, **kwargs)
+        _inputs, is_split_into_words, offset_mapping = self._args_parser(inputs, **kwargs)
+        kwargs["is_split_into_words"] = is_split_into_words
+        if is_split_into_words and not all(isinstance(input, list) for input in inputs):
+            return super().__call__([inputs], **kwargs)
         if offset_mapping:
             kwargs["offset_mapping"] = offset_mapping
 
         return super().__call__(inputs, **kwargs)
 
-    def preprocess(self, sentence, offset_mapping=None, **preprocess_params):
+    @overload
+    def preprocess(
+        self, sentence: str, is_split_into_words: bool = False, offset_mapping=None, **preprocess_params
+    ) -> Iterator[dict]: ...
+
+    @overload
+    def preprocess(
+        self, sentence: list[str], is_split_into_words: bool = True, offset_mapping=None, **preprocess_params
+    ) -> Iterator[dict]: ...
```

Comment:
The return type is the same in both cases so I'm not sure why we need the overload here!